src/app/(pages)/sign-up/page.tsx

```diff
@@ -1,138 +1,186 @@
 "use client";
-import { RegistersData } from '@/app/Data/fieldsData';
-
-import { Box, Button, FormControl,Typography,Divider,FormControlLabel, Checkbox,Link, TextField, InputAdornment } from '@mui/material';
-import { NextPage } from 'next';
-import { useForm, SubmitHandler } from 'react-hook-form';
-import { yupResolver } from '@hookform/resolvers/yup';
-import { RegisterSchema } from '@/app/Utils/AuthValidation'
-import { createUser } from '@/app/Service/AuthService';
-import { useRouter } from 'next/navigation';
-import { toast } from 'react-toastify';
-import FacebookIcon  from "@/Assets/icons8-facebook.svg";
+import { RegistersData } from "@/app/Data/fieldsData";
+
+import {
+  Box,
+  Button,
+  FormControl,
+  Typography,
+  Divider,
+  FormControlLabel,
+  Checkbox,
+  Link,
+  TextField,
+  InputAdornment,
+} from "@mui/material";
+import { NextPage } from "next";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { RegisterSchema } from "@/app/Utils/AuthValidation";
+import { createUser } from "@/app/Service/AuthServices";
+import { useRouter } from "next/navigation";
+import { toast } from "react-toastify";
+import FacebookIcon from "@/Assets/icons8-facebook.svg";
 import GoogleIcon from "@/Assets/icons8-google.svg";
 import Image from "next/image";
-import  styles from "./style.module.scss"
-import usePasswordVisibility from '@/app/Hooks/usePasswordVisibility';
-
+import styles from "./style.module.scss";
+import usePasswordVisibility from "@/app/Hooks/usePasswordVisibility";
 
 interface RegisterFormInputs {
   fullName: string;
   email: string;
   password: string;
 }
-const Page: NextPage = () =>{
-const {getPasswordInputProps} = usePasswordVisibility();
-
-const router = useRouter();
-  
-const { register, handleSubmit, formState: { errors} ,setError , reset } = useForm<RegisterFormInputs>({
-    resolver:yupResolver(RegisterSchema),
+const Page: NextPage = () => {
+  const { getPasswordInputProps } = usePasswordVisibility();
+
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+    reset,
+  } = useForm<RegisterFormInputs>({
+    resolver: yupResolver(RegisterSchema),
   });
-  const onSubmit: SubmitHandler<RegisterFormInputs> = async(data:RegisterFormInputs) =>{
-  
-    try{
-      const newUser = await createUser(data)
-      toast.success("Successful to Create Account")
-      reset()
-      router.push('/sign-in');
-      return newUser
-    }catch (error :any){
-      if(error.response.data.error==="Conflict"){
-        setError('email', {
-            type: 'manual',
-            message: "User Already Exists"});
-       }else{
-       toast.error(`Registration failed. Please try again`)
+  const onSubmit: SubmitHandler<RegisterFormInputs> = async (
+    data: RegisterFormInputs
```

Comment:
error handling here need some adjustment , so let's talk about that.
but use mutations